plugins/vendor/microsoft/teams/batch/KMicrosoftTeamsDropFolderEngine.php

```diff
@@ -0,0 +1,278 @@
+		$dropFolderFile->remoteId = $extendedItem['id'];
+		$dropFolderFile->name = $extendedItem['name'];
+		$dropFolderFile->contentUrl = $extendedItem['@microsoft.graph.downloadUrl'];
+		if (isset($extendedItem['description']))
+		{
+			$dropFolderFile->description = $extendedItem['description'];
+		}
+
+		$dropFolderFile->ownerId = $this->retrieveUserId($extendedItem['createdBy']);
+		$dropFolderFile->additionalUserIds = $this->retrieveParticipants($extendedItem);
+
+		try
+		{
+			$dropFolderFile = $this->dropFolderFileService->add($dropFolderFile);
+
+			$this->dropFolderFileService->updateStatus($dropFolderFile->id, KalturaDropFolderFileStatus::PENDING);
+			return $dropFolderFile;
+		}
+		catch(Exception $e)
+		{
+			KalturaLog::err('Cannot add new drop folder file with name ['.$extendedItem['id'].'] - '.$e->getMessage());
+			return null;
+		}
+
+	}
+
+	protected function retrieveParticipants($item)
+	{
+		$participants = array();
+
+		$callRecordId = $item['source']['externalId'];
+		$callRecord = $this->graphClient->getCallRecord($callRecordId);
+
+		if ($callRecord)
+		{
+			foreach ($callRecord['participants'] as $participant)
+			{
+				$userId = $participant['user']['id'];
+				$user = $this->graphClient->getUser($userId);
+				if ($user)
+				{
+					$participants[] = $user['mail'];
+				}
+			}
+		}
+
+		return implode(',', $participants);
+	}
+
+	protected function getUpdatedFileSize (KalturaDropFolderFile $dropFolderFile)
+	{
+		if (isset($this->singleRunFoundItems[$dropFolderFile->fileName]))
+		{
+			$item = $this->singleRunFoundItems[$dropFolderFile->fileName];
+			return $item['size'];
+		}
+
+		return null;
+	}
+
+	protected function retrieveUserId ($creatorInfo)
+	{
+		$userInfo = $creatorInfo['user'];
+		if(isset($userInfo['email']))
+		{
+			return $userInfo['email'];
+		}
+
+		return $userInfo['id'];
+	}
+
+	public function processFolder(KalturaBatchJob $job, KalturaDropFolderContentProcessorJobData $data)
+	{
+		KBatchBase::impersonate ($job->partnerId);
+
+		/* @var $data KalturaDropFolderContentProcessorJobData */
+		$dropFolder = $this->dropFolderPlugin->dropFolder->get ($data->dropFolderId);
+		//In the case of the microsoft teams drop folder engine, the only possible contentMatch policy is ADD_AS_NEW.
+		//Any other policy should cause an error.
+		switch ($data->contentMatchPolicy)
+		{
+			case KalturaDropFolderContentFileHandlerMatchPolicy::ADD_AS_NEW:
+				$this->addAsNewContent($job, $data, $dropFolder);
+				break;
+			default:
+				throw new kApplicativeException(KalturaDropFolderErrorCode::DROP_FOLDER_APP_ERROR, 'Content match policy not allowed for Microsoft Teams drop folders');
+				break;
+		}
+
+		KBatchBase::unimpersonate();
+	}
+
+	protected function addAsNewContent (KalturaBatchJob $job, KalturaDropFolderContentProcessorJobData $data, KalturaMicrosoftTeamsDropFolder $folder)
+	{
+		/* @var $data KalturaDropFolderContentProcessorJobData */
+		$resource = $this->getIngestionResource($job, $data);
```

Comment:
Need to indicate the source of this entry is TEAMS plugin by adding extension to entrySourceType in teams plugin

Reply:
@MosheMaorKaltura  according to Meirav, the same purpose can be accomplished by setting the entry's admin tags, as the Zoom plugin also does not add a new source type for entries.